access-grant/src/main/java/com/inrupt/client/accessgrant/Utils.java

```diff
@@ -44,6 +45,8 @@ public static Status asRevocationList2020(final Map<String, Object> credentialSt
                 idx = Integer.parseInt((String) index);
             } else if (index instanceof Integer) {
                 idx = (Integer) index;
+            } else if (index instanceof BigDecimal) {
```

Comment:
Why would an index value ever be larger than 100,000? As such a BigDecimal would not be needed. Plus, this is for index values, which are always integers. Even if we used very large numbers, a BigInteger would be the right value. Any non-integer value is invalid

Reply:
The `BigDecimal` class is instantiated by JSONB when parsing an integer from JSON (see the debug screenshot attached). This results in the index being unmodified, and left to its `-1` initialization value, if not checking for `BigDecimal`.

![Capture d’écran du 2023-06-21 10-35-48](https://github.com/inrupt/solid-client-java/assets/8236166/10fc5638-0ff4-4ebb-ac3c-85c7ec1dc7ac)